### PR TITLE
Add support for 'alwaysOutOfDate' PBX Shell Script property

### PR DESCRIPTION
**Pull request by marco-saia-datadog**
This PR adds support for the `alwaysOutOfDate` property in PBX Shell Scripts.

Setting `alwaysOutOfDate` to `true` disables dependency analysis for the given build phase, clearing warnings such as:

```sh
Script has ambiguous dependencies causing it to run on every build
```

Since by default the property is not added to the PBX project file by Xcode, I have decided to write it down in the PBX file only if explicitly specified in the `pbxShellScriptBuildPhaseObj` options, hence the null check.

### CODE REVIEW

**Comment by breautek** [comment]
I think this is a good addition, and one that Cordova itself could probably use. I've enabled the CI tests which obviously requires some chore work since it's still configured to use NodeJS 10 (It's been years since this package had updates!)

I've added a `3.1.0` milestone so that this PR isn't forgotten for when we prepare the next release, though a `4.0.0` already exists so this feature might be added with the next major release instead of a minor release.

Code wise, I think is :+1: just the CI needs to be updated to use modern NodeJS versions and then this PR will need a rebase.

**Comment by marco-saia-datadog** [comment]
Wonderful! Thank you @breautek 😃

**Comment by breautek** [comment]
Hey @marco-saia-datadog 

If you could rebase this PR it should fix the CI for you.

File: lib/pbxProject.js [hunk 1]
```diff
@@ -1644,6 +1644,10 @@ function pbxShellScriptBuildPhaseObj(obj, options, phaseName) {
     obj.shellPath = options.shellPath;
     obj.shellScript = '"' + options.shellScript.replace(/"/g, '\\"') + '"';
 
+    if (options.alwaysOutOfDate != null) {
```
Comment (breautek) [inline lib/pbxProject.js:1647]: The current codebase uses the triple equality and we should continue this pattern for consistency.

This will more strictly test against `null` without type coercion (though functionality I believe in this case it isn't any different than `!=`).

```suggestion
    if (options.alwaysOutOfDate !== null) {
```

File: test/addBuildPhase.js [hunk 2]
```diff
@@ -194,4 +194,26 @@ exports.addBuildPhase = {
...
+        var options = {
+            shellPath: '/bin/sh',
+            shellScript: 'test',
+            alwaysOutOfDate: true
+        };
+
+        var buildPhase = proj.addBuildPhase([], 'PBXShellScriptBuildPhase', 'Run a script', proj.getFirstTarget().uuid, options).buildPhase;
+        test.equal(buildPhase.shellPath, '/bin/sh');
+        test.equal(buildPhase.shellScript, '"test"');
+        test.equal(buildPhase.alwaysOutOfDate, true);
```
Comment (breautek) [inline test/addBuildPhase.js:207]: The way this test is conducted this will pass but was this tested against a real project?

In my xcode projects, xcode will use `1` or `0` values instead of true and false. I'm not sure if our writer will translate booleans to 0/1 when printed to file, or if it will print as "true", I'm also not sure if xcode will accept a "true" value over `1`.

For the sake of "self-documenting" maybe the test should match the value types that is used by xcode?

**Review by breautek — COMMENTED** [review]
Didn't test this PR explicitly but did confirm that `alwaysOutOfDate` is a field that XCode (At least XCode 15 uses). :+1: 

`alwaysOutOfDate` is only ever assigned if the options value is not null. This means if the value is `undefined`, then `obj.alwaysOutOfDate` will also be undefined, matching old behaviour. If the value is explicitly set to `null`, then it will be left `undefined` on `obj.alwaysOutOfDate`. Any other value is passed through. :+1:

I left a nitpick to use `!==` over `!=` for mostly code consistency reasons.

I do have concerns about the test using boolean true when Xcode appears to use `1`/`0` values and I don't see any evidence that our [pbxWriter](https://github.com/apache/cordova-node-xcode/blob/master/lib/pbxWriter.js#L263) will do this translation.

File: test/addBuildPhase.js (hunk 2 above)
Comment (breautek) [reply test/addBuildPhase.js:207]: Also other places in the test will use the integers to represent the boolean as well: https://github.com/apache/cordova-node-xcode/blob/master/test/addBuildPhase.js#L61 so using ints will also make the test more consistent with other parts of the codebase, even if `true`/`false` is accepted.

**Comment by KiwiKilian** [comment]
Thanks for working on this! Any chance to get this moving forward? Any kind of support possible? This would be a very helpful improvement to the Cocoapods and React Native ecosystem.

**Comment by erisu** [comment]
The changes from this PR were cherry-picked, rebased, and merged under PR #173.
Thank you for your contribution.

**Comment by marco-saia-datadog** [comment]
Thank you @erisu 🙌
